### PR TITLE
[1.1.1.1] Add Oblivious DNS over HTTPS

### PR DESCRIPTION
**Pull request by RebeccaTamachiro**
### Summary

PCX-7677

### Documentation checklist

- [ ] The [documentation style guide](https://developers.cloudflare.com/style-guide/) has been adhered to.

### CODE REVIEW

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx [hunk 1]
```diff
@@ -7,9 +7,11 @@ sidebar:
   label: Oblivious DoH
 ---
 
-As announced on [our blog](https://blog.cloudflare.com/oblivious-dns/), since late 2020, Cloudflare 1.1.1.1 supports Oblivious DNS over HTTPS (ODoH) ([RFC 9230](https://www.rfc-editor.org/rfc/rfc9230.html)).
+As announced on [our blog](https://blog.cloudflare.com/oblivious-dns/), since late 2020, Cloudflare 1.1.1.1 supports Oblivious DNS over HTTPS (ODoH).
 
-ODoH is a protocol for performing remote Domain Name System (DNS) resolution based on the Oblivious HTTP protocol (OHTTP) ([RFC 9458](https://www.rfc-editor.org/rfc/rfc9458.html)).
+:::caution
+Although ODoH is defined in [RFC 9230](https://www.rfc-editor.org/rfc/rfc9230.html), it is experimental. 1.1.1.1 supports ODoH as [described bellow](#cloudflare-and-third-party-products), but this protocol is not endorsed by IETF nor by Cloudflare.
```
Comment (xofyarg) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx]: I think we only need to mention the RFC is experimental, and is not endorsed by IETF, as stated on the right side bar here: https://datatracker.ietf.org/doc/html/rfc9230

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx [hunk 2]
```diff
@@ -31,7 +33,7 @@ Cloudflare 1.1.1.1 supports ODoH by acting as a target that can be reached at `o
 
 At launch, a few proxy partners included [PCCW](https://www.pccw.com/), [SURF](https://www.surf.nl/), and [Equinix](https://www.equinix.com/).
 
-Finally, open source test clients are available in [Rust](https://github.com/cloudflare/odoh-client-rs) or [Go](https://github.com/cloudflare/odoh-client-go).
+Finally, open source test clients are available in [Rust](https://github.com/DNSCrypt/dnscrypt-resolvers/blob/master/v3/odoh-relays.md) or [Go](https://github.com/cloudflare/odoh-client-go).
```
Comment (xofyarg) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx]: `dnscrypt` is not a rust library, and the standalone client is dnscrypt-proxy: https://github.com/DNSCrypt/dnscrypt-proxy.

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+As announced on [our blog](https://blog.cloudflare.com/oblivious-dns/), since late 2020, Cloudflare 1.1.1.1 supports Oblivious DNS over HTTPS (ODoH).
+
+:::caution
+Although ODoH is defined in [RFC 9230](https://www.rfc-editor.org/rfc/rfc9230.html), it is experimental. 1.1.1.1 supports ODoH as [described bellow](#cloudflare-and-third-party-products), but this protocol is not endorsed by IETF nor by Cloudflare.
+:::
+
+## How ODoH works
+
+OHTTP improves privacy by separating the contents of an HTTP request (and response) from its requester IP address. To achieve this in DNS resolution, a proxy and a target are introduced between the client and the upstream DNS resolver:
```
Comment (xofyarg) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx]: ODoH doesn't quite related to OHTTP, in fact it doesn't rely on HTTP to transfer data.

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+- The target only has access to the encrypted query and the proxy's IP address.
```
Comment (xofyarg) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx]: It is emphasized below that the proxy has no visibility to the DNS message, so I think we should also emphasized here that the target has no visibility to the client's IP address.

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+Cloudflare 1.1.1.1 supports ODoH by acting as a target that can be reached at `odoh.cloudflare-dns.com`.
+
+At launch, a few proxy partners included [PCCW](https://www.pccw.com/), [SURF](https://www.surf.nl/), and [Equinix](https://www.equinix.com/).
+
+Finally, open source clients such as [dnscrypt-proxy](https://github.com/DNSCrypt/dnscrypt-proxy) can be used to make ODoH queries.
+
+## Related resources
+
+- [Proving Oblivious HTTP privacy properties](https://blog.cloudflare.com/stronger-than-a-promise-proving-oblivious-http-privacy-properties/) blog post
```
Comment (xofyarg) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx]: I guess this is not quite related. Maybe put https://blog.cloudflare.com/hybrid-public-key-encryption/ here, which is used by ODoH?

**Review by xofyarg — APPROVED** [review]
LGTM, thanks!

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+This means that, as long as the proxy and the target do not collude, no single entity can have access to both the DNS messages and the client IP address at the same time. Also, clients are in complete control of proxy and target selection.
+
+Additionally, clients encrypt their query for the target using Hybrid Public Key Encryption (HPKE). A target's public key is obtained via DNS, where it is bundled into an HTTPS resource record and protected by DNSSEC.
+
+## Cloudflare and third-party products
+
+Cloudflare 1.1.1.1 supports ODoH by acting as a target that can be reached at `odoh.cloudflare-dns.com`.
+
+At launch, a few proxy partners included [PCCW](https://www.pccw.com/), [SURF](https://www.surf.nl/), and [Equinix](https://www.equinix.com/).
```
Comment (Oxyjun) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx]: Just for my own understanding: are they still included? "At launch" + "included" (past tense) made me a bit unsure if we're saying we still include them or not 😄

**Review by Oxyjun — APPROVED** [review]

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx (hunk 6 above)
Comment (vendemiat) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx]: needs rewording.

File: src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+- The target only has access to the encrypted query and the proxy's IP address, while not having visibility over the client's IP address.
+
+- Only the intended target can read the content of the query and produce a response, which is also encrypted.
+
+This means that, as long as the proxy and the target do not collude, no single entity can have access to both the DNS messages and the client IP address at the same time. Also, clients are in complete control of proxy and target selection.
+
+Additionally, clients encrypt their query for the target using Hybrid Public Key Encryption (HPKE). A target's public key is obtained via DNS, where it is bundled into an HTTPS resource record and protected by DNSSEC.
+
+## Cloudflare and third-party products
```
Comment (vendemiat) [inline src/content/docs/1.1.1.1/encryption/oblivious-dns-over-https.mdx:30]: Can we include Apple private relay?

**Review by vendemiat — APPROVED** [review]

**Pull request merged**